File: 209_Minimum_Size_Subarray_Sum.py
class Solution(object):
    def minSubArrayLen(self, s, nums):
        """
        :type s: int
        :type nums: List[int]
        :rtype: int
        """
        
        i = 0
        j = 0
        ans = len(nums)+1
        sums = 0
        
        while j>=i:
            if sums >= s:
                ans = min(ans, j-i)
                sums -= nums[i]
                i += 1
            else:
                if j>=len(nums):
                    break;
                sums += nums[j]
                j += 1
            
        if ans > len(nums):
            return 0
        else:
            return ans

File: 209_Minimum_Size_Subarray_Sum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        
        int i=0, j=0, sum=0, ans=nums.size()+1;
        
        while(i<=j){
            if(sum>=s){
                ans = (ans<j-i)? ans:j-i;
                sum -= nums[i];
                i++;
            }
            else{
                if(j>=nums.size())
                    break;
                sum += nums[j];
                j++;
            }
        }
        
        if(ans>nums.size())
            return 0;
        else
            return ans;
                
    }
};
